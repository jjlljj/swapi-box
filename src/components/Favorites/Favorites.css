.favorites {
  min-height: 80vh
}

.favorites h3 {
  align-self: center;
  font-size: 40px;
  color: #f2f2f2;
  margin: 30px;
}