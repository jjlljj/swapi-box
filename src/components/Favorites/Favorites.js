import React from 'react';
import { arrayOf, func, object } from 'prop-types';
import './Favorites.css';
import Card from '../Card/Card';

const Favorites = ({ cards, addToFav }) => {

  const renderedCards = cards.map(card => {
    return (<Card card={card} addToFav={addToFav} key={card.name}/>);
  });

  const noFavorites = ( <h3 className="no-fav">No Favorites Found</h3> );

  const rendered = cards.length ? renderedCards : noFavorites;

  return (
    <div className="card-container"> 
      {
        rendered
      }
    </div>
  );
};

Favorites.propTypes = {
  addToFav: func,
  cards: arrayOf(object)
};

export default Favorites;
