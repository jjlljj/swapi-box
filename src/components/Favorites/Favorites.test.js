/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Favorites from './Favorites';

describe('Favorites', () => {

  let renderedComponent
  let mockAddFavorite
  const mockCards = [
    {
      Climate: "murky",
      Population: "unknown",
      Residents: "",
      Terrain: "swamp, jungles",
      cardType: "planets",
      favorite: true,
      name: "Dagobah"
    },
    {
      Homeword: "Tatooine",
      Population: "200000",
      Species: "Human",
      cardType: "people",
      name: "Beru Whitesun lars"
    }
  ]

  beforeEach(()=> {
    mockAddFavorite = jest.fn()
    renderedComponent = shallow(<Favorites cards={mockCards} addToFav={mockAddFavorite} />)
  })

  it('should match snapshot when passed cards', () => {
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should match snapshot when passed only one card', () => {
    const mockCard = [
      {
        Climate: "temperate, tropical", 
        Population: "1000",
        Residents: "",
        Terrain: "jungle, rainforests",
        cardType: "planets",
        favorite: true,
        name: "Yavin IV"
      }
    ]

    renderedComponent = shallow(<Favorites cards={mockCard} addToFav={mockAddFavorite} />)

    expect(renderedComponent).toMatchSnapshot()
  })

})
