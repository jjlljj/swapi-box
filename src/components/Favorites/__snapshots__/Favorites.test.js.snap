// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot when passed cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    addToFav={[Function]}
    cards={
        Array [
            Object {
              "Climate": "murky",
              "Population": "unknown",
              "Residents": "",
              "Terrain": "swamp, jungles",
              "cardType": "planets",
              "favorite": true,
              "name": "Dagobah",
            },
            Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "cardType": "people",
              "name": "Beru Whitesun lars",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFav={[Function]}
          card={
                    Object {
                              "Climate": "murky",
                              "Population": "unknown",
                              "Residents": "",
                              "Terrain": "swamp, jungles",
                              "cardType": "planets",
                              "favorite": true,
                              "name": "Dagobah",
                            }
          }
/>,
        <Card
          addToFav={[Function]}
          card={
                    Object {
                              "Homeword": "Tatooine",
                              "Population": "200000",
                              "Species": "Human",
                              "cardType": "people",
                              "name": "Beru Whitesun lars",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Dagobah",
        "nodeType": "function",
        "props": Object {
          "addToFav": [Function],
          "card": Object {
            "Climate": "murky",
            "Population": "unknown",
            "Residents": "",
            "Terrain": "swamp, jungles",
            "cardType": "planets",
            "favorite": true,
            "name": "Dagobah",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Beru Whitesun lars",
        "nodeType": "function",
        "props": Object {
          "addToFav": [Function],
          "card": Object {
            "Homeword": "Tatooine",
            "Population": "200000",
            "Species": "Human",
            "cardType": "people",
            "name": "Beru Whitesun lars",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFav={[Function]}
            card={
                        Object {
                                    "Climate": "murky",
                                    "Population": "unknown",
                                    "Residents": "",
                                    "Terrain": "swamp, jungles",
                                    "cardType": "planets",
                                    "favorite": true,
                                    "name": "Dagobah",
                                  }
            }
/>,
          <Card
            addToFav={[Function]}
            card={
                        Object {
                                    "Homeword": "Tatooine",
                                    "Population": "200000",
                                    "Species": "Human",
                                    "cardType": "people",
                                    "name": "Beru Whitesun lars",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Dagobah",
          "nodeType": "function",
          "props": Object {
            "addToFav": [Function],
            "card": Object {
              "Climate": "murky",
              "Population": "unknown",
              "Residents": "",
              "Terrain": "swamp, jungles",
              "cardType": "planets",
              "favorite": true,
              "name": "Dagobah",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Beru Whitesun lars",
          "nodeType": "function",
          "props": Object {
            "addToFav": [Function],
            "card": Object {
              "Homeword": "Tatooine",
              "Population": "200000",
              "Species": "Human",
              "cardType": "people",
              "name": "Beru Whitesun lars",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Favorites should match snapshot when passed only one card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    addToFav={[Function]}
    cards={
        Array [
            Object {
              "Climate": "temperate, tropical",
              "Population": "1000",
              "Residents": "",
              "Terrain": "jungle, rainforests",
              "cardType": "planets",
              "favorite": true,
              "name": "Yavin IV",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFav={[Function]}
          card={
                    Object {
                              "Climate": "temperate, tropical",
                              "Population": "1000",
                              "Residents": "",
                              "Terrain": "jungle, rainforests",
                              "cardType": "planets",
                              "favorite": true,
                              "name": "Yavin IV",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Yavin IV",
        "nodeType": "function",
        "props": Object {
          "addToFav": [Function],
          "card": Object {
            "Climate": "temperate, tropical",
            "Population": "1000",
            "Residents": "",
            "Terrain": "jungle, rainforests",
            "cardType": "planets",
            "favorite": true,
            "name": "Yavin IV",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFav={[Function]}
            card={
                        Object {
                                    "Climate": "temperate, tropical",
                                    "Population": "1000",
                                    "Residents": "",
                                    "Terrain": "jungle, rainforests",
                                    "cardType": "planets",
                                    "favorite": true,
                                    "name": "Yavin IV",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Yavin IV",
          "nodeType": "function",
          "props": Object {
            "addToFav": [Function],
            "card": Object {
              "Climate": "temperate, tropical",
              "Population": "1000",
              "Residents": "",
              "Terrain": "jungle, rainforests",
              "cardType": "planets",
              "favorite": true,
              "name": "Yavin IV",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
